lib/components/ToggleButton/ToggleButton.spec.jsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from 'react';
+import { mount, render, shallow } from 'enzyme';
+import { ToggleButton } from './ToggleButton';
+import { act } from 'react-dom/test-utils';
+
+describe('<ToggleButton />', () => {
+	it('should not throw', () =>
+		expect(() => shallow(<ToggleButton />)).not.toThrow());
+
+	it('should match snapshot without props', () => {
+		expect(render(<ToggleButton />)).toMatchSnapshot();
+	});
+
+	it('should match snapshot when un-toggled', () => {
+		expect(render(<ToggleButton toggled={false} />)).toMatchSnapshot();
+	});
+
+	it('should match snapshot when un-toggled and disabled', () => {
+		expect(
+			render(<ToggleButton toggled={false} disabled={true} />)
+		).toMatchSnapshot();
+	});
+
+	it('should match snapshot when toggled', () => {
+		expect(render(<ToggleButton toggled={true} />)).toMatchSnapshot();
+	});
+
+	it('should match snapshot when toggled and disabled', () => {
+		expect(
+			render(<ToggleButton toggled={true} disabled={true} />)
+		).toMatchSnapshot();
+	});
+
+	it('should pass on className to dom element', () => {
+		const toggleButton = mount(
+			<ToggleButton className="toggleButton-class" value={10} />
+		);
+		expect(
+			toggleButton.find('div.toggleButton-class').length === 1
+		).toBeTruthy();
+	});
+
+	it('should set toggle to false by default', () => {
+		const toggleButton = mount(<ToggleButton />);
+		expect(toggleButton.find('.toggled').length === 0).toBeTruthy();
+	});
+
+	it('should be toggled on when toggled prop is set to true', () => {
+		const toggleButton = mount(<ToggleButton toggled={true} />);
+		expect(toggleButton.find('.toggled').length === 1).toBeTruthy();
+	});
+
+	it('should be enabled by default', () => {
+		const toggleButton = mount(<ToggleButton />);
+		expect(toggleButton.find('[disabled]').length === 0).toBeTruthy();
+	});
+
+	it('should be disabled on when disabled prop is set to true', () => {
+		const toggleButton = mount(<ToggleButton disabled={true} />);
+		expect(toggleButton.find('[disabled]').length === 1).toBeTruthy();
+	});
+
+	it('should fire change with the correct changed value when clicked', () => {
+		const spyedCallback = jest.fn();
+
+		const wrapper = mount(
+			<ToggleButton onChange={spyedCallback} toggled={false} />
+		);
+
+		wrapper.find('div').simulate('click');
+
+		expect(spyedCallback).toHaveBeenCalledWith(true);
+
+		wrapper.find('div').simulate('click');
+
+		expect(spyedCallback).toHaveBeenCalledWith(false);
+
+		wrapper.find('div').simulate('click');
+
+		expect(spyedCallback).toHaveBeenCalledWith(true);
+		expect(spyedCallback).toHaveBeenCalledTimes(3);
+
+		wrapper.unmount();
+	});
+
+	it('should fire change if clicked while disabled', () => {
```

Comment:
Probably mean should *not*?

```suggestion
	it('should not fire change if clicked while disabled', () => {
```